src/service/TransactionService.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Observable} from 'rxjs';
+import { mergeMap, toArray} from 'rxjs/operators';
+import { ReceiptHttp } from '../infrastructure/ReceiptHttp';
+import { TransactionHttp } from '../infrastructure/TransactionHttp';
+import { Address } from '../model/account/Address';
+import { MosaicId } from '../model/mosaic/MosaicId';
+import { NamespaceId } from '../model/namespace/NamespaceId';
+import { ResolutionType } from '../model/receipt/ResolutionType';
+import { Statement } from '../model/receipt/Statement';
+import { Transaction } from '../model/transaction/Transaction';
+import { ITransactionService } from './interfaces/ITransactionService';
+
+/**
+ * Transaction Service
+ */
+export class TransactionService implements ITransactionService {
+
+    private readonly transactionHttp: TransactionHttp;
+    private readonly receiptHttp: ReceiptHttp;
+    /**
+     * Constructor
+     * @param url Base catapult-rest url
+     */
+    constructor(url: string) {
+        this.transactionHttp = new TransactionHttp(url);
+        this.receiptHttp = new ReceiptHttp(url);
+    }
+
+    /**
+     * @internal
+     * Extract resolved address | mosaic from block receipt
+     * @param resolutionType Resolution type: Address / Mosaic
+     * @param unresolved Unresolved address / mosaicId
+     * @param statement Block receipt statement
+     * @param transactionIndex Transaction index
+     * @param height Transaction height
+     * @param aggregateTransactionIndex Transaction index for aggregate
+     * @returns {MosaicId | Address}
+     */
+    public static getResolvedFromReceipt(resolutionType: ResolutionType,
```

Comment:
I'll split this method in 2, getResolvedAddressFromRecipient and getResolvedMosaicIdFromRecipient

Reply:
Can do but will probably create blocks of duplicated codes ??

Reply:
you can have a third method that does the filtering primary/secondary filtering. I'll push the java example

Reply:
outdated